torchvision/csrc/cpu/image/readjpeg_cpu.cpp

```diff
@@ -0,0 +1,148 @@
+#include "readjpeg_cpu.h"
+
+#include <ATen/ATen.h>
+#include <setjmp.h>
+#include <string>
+
+#if !JPEG_FOUND
+
+torch::Tensor decodeJPEG(const torch::Tensor& data) {
+  AT_ERROR("decodeJPEG: torchvision not compiled with libjpeg support");
+}
+
+#else
+#include <jpeglib.h>
+
+const static JOCTET EOI_BUFFER[1] = {JPEG_EOI};
+char jpegLastErrorMsg[JMSG_LENGTH_MAX];
```

Comment:
for a next PR: it would be great if there was a way of avoiding having those globals.